apps/desktop-wallet/src/components/AssetBadge.tsx

```diff
@@ -53,7 +53,7 @@ const AssetBadge = ({ assetId, amount, simple, className }: AssetBadgeProps) =>
       {nftInfo?.name ? (
         <AssetSymbol>{nftInfo?.name}</AssetSymbol>
       ) : amount !== undefined ? (
-        <Amount value={amount} suffix={assetInfo?.symbol} />
+        <Amount value={amount} suffix={assetInfo?.symbol} decimals={assetInfo?.decimals} />
```

Comment:
Optional parameters backfire quite often. Could use explicit parameters for important ones, in this case, we could introduce `<AssetAmount mandatoryDecimals=.../>`

Reply:
Agreed. The `Amount` component is due for a cleanup. I think we can do better with clever types. Just an idea :
- Expecting a mandatory amount type prop to be either `fiat` or `asset`
- If it's asset, then the `decimals` field is mandatory.